src/ActionModal/utils/networkMessages/polkadot-testnet.js

```diff
@@ -9,132 +10,43 @@ export async function MsgSend(
-//       delegator_address: senderAddress,
-//       amount: Coin({ amount, denom })
-//     }
-//   }
-// }
-
-// export function MsgRedelegate(
-//   senderAddress,
-//   { validatorSourceAddress, validatorDestinationAddress, amount, denom }
-// ) {
-//   /* istanbul ignore next */
-//   return {
-//     type: `cosmos-sdk/MsgBeginRedelegate`,
-//     value: {
-//       delegator_address: senderAddress,
-//       validator_src_address: validatorSourceAddress,
-//       validator_dst_address: validatorDestinationAddress,
-//       amount: Coin({ amount, denom })
-//     }
-//   }
-// }
-
-// // Governance
-// export function MsgSubmitProposal(
-//   senderAddress,
-//   {
-//     title,
-//     description,
-//     initialDeposits // [{ denom, amount }]
-//   }
-// ) {
-//   /* istanbul ignore next */
-//   return {
-//     type: `cosmos-sdk/MsgSubmitProposal`,
-//     value: {
-//       content: {
-//         type: "cosmos-sdk/TextProposal",
-//         value: {
-//           title,
-//           description
-//         }
-//       },
-//       proposer: senderAddress,
-//       initial_deposit: initialDeposits.map(Coin)
-//     }
-//   }
-// }
-
-// export function MsgVote(senderAddress, { proposalId, option }) {
-//   /* istanbul ignore next */
-//   return {
-//     type: `cosmos-sdk/MsgVote`,
-//     value: {
-//       voter: senderAddress,
-//       proposal_id: proposalId,
-//       option
-//     }
-//   }
-// }
-
-// export function MsgDeposit(
-//   senderAddress,
-//   {
-//     proposalId,
-//     amounts // [{ denom, amount }]
-//   }
-// ) {
-//   /* istanbul ignore next */
-//   return {
-//     type: `cosmos-sdk/MsgDeposit`,
-//     value: {
-//       depositor: senderAddress,
-//       proposal_id: proposalId,
-//       amount: amounts.map(Coin)
-//     }
-//   }
-// }
-
-// export function MsgWithdrawDelegationReward(
-//   senderAddress,
-//   { validatorAddress }
-// ) {
-//   /* istanbul ignore next */
-//   return {
-//     type: `cosmos-sdk/MsgWithdrawDelegationReward`,
-//     value: {
-//       delegator_address: senderAddress,
-//       validator_address: validatorAddress
-//     }
-//   }
-// }
-
-// function Coin({ amount, denom }) {
-//   return {
-//     amount: String(amount),
-//     denom
-//   }
-// }
+// Staking
+export async function MsgDelegate(senderAddress, { validatorAddress, amount }) {
+  // stake with all existing plus the selected
+  const api = await getAPI()
+  const response = await api.query.staking.nominators(senderAddress)
+  const { targets: delegatedValidators = [] } = response.toJSON() || {}
+  const transactions = []
+
+  // Check if controller is already set
+  const controller = await api.query.staking.bonded(senderAddress)
+
+  if (controller.toString() === `` && amount > 0) {
+    const payee = 0
+    transactions.push(await api.tx.staking.bond(senderAddress, amount, payee))
+  } else {
+    transactions.push(await api.tx.staking.bondExtra(amount))
```

Comment:
"Schedule a portion of the stash to be unlocked ready for transfer out after the bond period ends" (tx.ts file in @polkadot/api)

When does the bond period end? 👀

Reply:
`unlocked` sounds like unbonding. did you get this from the description for bond_extra?

Reply:
yes, from bond_extra

Reply:
mmm description for `bond_extra` is:

```
Add some extra amount that have appeared in the stash `free_balance` into the balance up
 for staking.

 Use this if there are additional funds in your stash account that you wish to bond.
 Unlike [`bond`] or [`unbond`] this function does not impose any limitation on the amount
 that can be added.

 The dispatch origin for this call must be _Signed_ by the stash, not the controller.

 Emits `Bonded`.
```

Makes more sense to me ^^